packages/python-packages/apiview-stub-generator/tests/class_parsing_test.py

```diff
@@ -65,231 +81,388 @@ def test_class_with_decorators(self):
         _check_all(actuals, expected, obj)
 
     def test_overloads(self):
         obj = SomethingWithOverloads
-        class_node = ClassNode(name=obj.__name__, namespace=obj.__name__, parent_node=None, obj=obj, pkg_root_namespace=self.pkg_namespace)
-        lines = _render_lines(_tokenize(class_node))
+        class_node = ClassNode(
+            name=obj.__name__,
+            namespace=obj.__name__,
+            parent_node=None,
+            obj=obj,
+            pkg_root_namespace=self.pkg_namespace,
+            apiview=MockApiView,
+        )
+        tokens = _tokenize(class_node)
+        lines = _render_lines(tokens)
         assert lines[2].lstrip() == "@overload"
         actual1 = _merge_lines(lines[3:10])
-        expected1 = 'def double(self, input: int = 1, *, test: bool = False, **kwargs) -> int'
+        expected1 = "def double(self, input: int = 1, *, test: bool = False, **kwargs) -> int"
         _check(actual1, expected1, SomethingWithOverloads)
 
         assert lines[11].lstrip() == "@overload"
         actual2 = _merge_lines(lines[12:19])
-        expected2 = 'def double(self, input: Sequence[int] = [1], *, test: bool = False, **kwargs) -> list[int]'
+        expected2 = "def double(self, input: Sequence[int] = [1], *, test: bool = False, **kwargs) -> list[int]"
         _check(actual2, expected2, SomethingWithOverloads)
 
         actual3 = _merge_lines(lines[20:27])
-        expected3 = 'def double(self, input: int | Sequence[int], *, test: bool = False, **kwargs) -> int | list[int]'
+        expected3 = "def double(self, input: int | Sequence[int], *, test: bool = False, **kwargs) -> int | list[int]"
         _check(actual3, expected3, SomethingWithOverloads)
 
         assert lines[28].lstrip() == "@overload"
         actual4 = _merge_lines(lines[29:35])
-        expected4 = 'def something(self, id: str, *args, **kwargs) -> str'
+        expected4 = "def something(self, id: str, *args, **kwargs) -> str"
         _check(actual4, expected4, SomethingWithOverloads)
 
         assert lines[36].lstrip() == "@overload"
         actual5 = _merge_lines(lines[37:43])
-        expected5 = 'def something(self, id: int, *args, **kwargs) -> str'
+        expected5 = "def something(self, id: int, *args, **kwargs) -> str"
         _check(actual5, expected5, SomethingWithOverloads)
 
         actual6 = _merge_lines(lines[44:])
-        expected6 = 'def something(self, id: int | str, *args, **kwargs) -> str'
+        expected6 = "def something(self, id: int | str, *args, **kwargs) -> str"
         _check(actual6, expected6, SomethingWithOverloads)
 
+        # If the token is an overload decorator, it should have the correct RelatedToLine.
+        for idx, token in enumerate(tokens[0]["Children"]):
+            if len(token["Tokens"]) > 0 and token["Tokens"][0]["Value"] == "@overload":
+                assert "RelatedToLine" in token
```

Comment:
RelatedToLine tests - may be better to go through all of them and count, rather than just checking a specific one.